platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/builders/internal/StandardTaskSchedulerBuilder.java

```diff
@@ -0,0 +1,178 @@
+import com.swirlds.common.metrics.extensions.NoOpFractionalTimer;
+import com.swirlds.common.metrics.extensions.StandardFractionalTimer;
+import com.swirlds.common.wiring.counters.ObjectCounter;
+import com.swirlds.common.wiring.model.StandardWiringModel;
+import com.swirlds.common.wiring.schedulers.TaskScheduler;
+import com.swirlds.common.wiring.schedulers.builders.TaskSchedulerType;
+import com.swirlds.common.wiring.schedulers.internal.ConcurrentTaskScheduler;
+import com.swirlds.common.wiring.schedulers.internal.DirectTaskScheduler;
+import com.swirlds.common.wiring.schedulers.internal.NoOpTaskScheduler;
+import com.swirlds.common.wiring.schedulers.internal.SequentialTaskScheduler;
+import com.swirlds.common.wiring.schedulers.internal.SequentialThreadTaskScheduler;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Objects;
+import java.util.concurrent.ForkJoinPool;
+
+/**
+ * A builder for {@link TaskScheduler}s.
+ *
+ * @param <OUT> the output type of the primary output wire for this task scheduler (use {@link Void} for a scheduler with
+ *            no output)
+ */
+public class StandardTaskSchedulerBuilder<OUT> extends AbstractTaskSchedulerBuilder<OUT> {
+
+    /**
+     * Constructor.
+     *
+     * @param platformContext the platform context
+     * @param model           the wiring model
+     * @param name            the name of the task scheduler. Used for metrics and debugging. Must be unique. Must only
+     *                        contain alphanumeric characters and underscores.
+     * @param defaultPool     the default fork join pool, if none is provided then this pool will be used
+     */
+    public StandardTaskSchedulerBuilder(
+            @NonNull final PlatformContext platformContext,
+            @NonNull final StandardWiringModel model,
+            @NonNull final String name,
+            @NonNull final ForkJoinPool defaultPool) {
+
+        super(platformContext, model, name, defaultPool);
+    }
+
+    /**
+     * Build a busy timer if enabled.
+     *
+     * @return the busy timer, or null if not enabled
+     */
+    @NonNull
+    private FractionalTimer buildBusyTimer() {
+        if (!busyFractionMetricEnabled || type == NO_OP) {
+            return NoOpFractionalTimer.getInstance();
+        }
+        if (type == TaskSchedulerType.CONCURRENT) {
+            throw new IllegalStateException("Busy fraction metric is not compatible with concurrent schedulers");
+        }
+        return new StandardFractionalTimer(platformContext.getTime());
+    }
+
+    /**
+     * Register all configured metrics.
+     *
+     * @param unhandledTaskCounter the counter that is used to track the number of scheduled tasks
+     */
+    private void registerMetrics(
+            @Nullable final ObjectCounter unhandledTaskCounter, @NonNull final FractionalTimer busyFractionTimer) {
+
+        if (type == NO_OP) {
```

Comment:
looks like we handle NO_OP types correctly anyway by returning a no-op task scheduler. It's not clear to me why we trivially return here considering we should be treating a no-op scheduler as a 'normal' scheduler.